nautobot/extras/filters/__init__.py

```diff
@@ -1104,6 +1112,51 @@ class Meta:
         fields = ("id",)
 
 
+#
+# StaticGroups
+#
+
+
+class StaticGroupFilterSet(NautobotFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "name": "icontains",
+            "description": "icontains",
+            "content_type__app_label": "icontains",
+            "content_type__model": "icontains",
+        },
+    )
+    content_type = ContentTypeMultipleChoiceFilter(choices=StaticGroupModelsQuery().get_choices, conjoined=False)
+
+    class Meta:
+        model = StaticGroup
+        fields = "__all__"
+
+
+class StaticGroupAssociationFilterSet(NautobotFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "static_group__name": "icontains",
+            "static_group__description": "icontains",
+            "associated_object_type__app_label": "icontains",
+            "associated_object_type__model": "icontains",
+        }
+    )
+
+    static_group = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=StaticGroup.objects.all(),
+        to_field_name="name",
+        label="Static group (name or PK)",
```

Comment:
We might want to stick with ID to be consistent. We have about 200 filters using ID in the label and only 2 using PK.

```suggestion
        label="Static group (name or ID)",
```